### PR TITLE
Start creating a scoped session object in the models module

### PR DESCRIPTION
**Pull request by jeremycline**
Since we load the configuration at module import time anyway, there's no
reason we can't create the database engine at import time as well. The
engine only connects to the database when connect is explicitly called
or something that requires a connection is called[0]. Furthermore, we
can also create the scoped session object.

This adds the ability to create the database to the fmn-createdb script
and stops that from calling ``fmn.lib.models.init``. That function has
been deprecated.

[0] http://docs.sqlalchemy.org/en/latest/core/engines.html#sqlalchemy.create_engine

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

### CODE REVIEW

File: fmn/lib/db.py [hunk 1]
```diff
@@ -20,50 +20,68 @@
 """
 from __future__ import print_function, unicode_literals
 
-import sys
+import argparse
```
Comment (bowlofeggs) [inline fmn/lib/db.py:23]: I would recommend [click](http://click.pocoo.org/5/) over argparse in general.

File: fmn/lib/models.py [hunk 2]
```diff
@@ -70,28 +81,25 @@ def init(db_url, alembic_ini=None, debug=False, create=False):
...
+            information with regards to the database engine, the host to
+            connect to, the user and password and the database name.
+            ie: <engine>://<user>:<password>@<host>/<dbname>
+        alembic_ini (str): path to the alembic ini file. This is necessary
+            to be able to use alembic correctly, but not for the unit-tests.
+        debug (bool): a boolean specifying wether we should have the verbose
+            output of sqlalchemy or not.
+
+    Returns:
+        scopedsession: An SQLAlchemy scoped session.
```
Comment (bowlofeggs) [inline fmn/lib/models.py:98]: This docblock change seems unrelated to the purpose of the rest of this commit. I recommend splitting it into a separate commit.

**Review by bowlofeggs — APPROVED** [review]
Just a couple suggestions at your option. LGTM!

File: fmn/lib/db.py (hunk 1 above)
Comment (pypingou) [reply fmn/lib/db.py:23]: one is in stdlib and the other adds a new dependency, so there is a tradeoff :)

Comment (jeremycline) [reply fmn/lib/db.py:23]: Yeah, click looks good, but for this (which is only meant to create the database) isn't going to be a complex CLI. If it starts to grow, I'll reconsider using click!
